python/lsst/daf/butler/core/datasets/ref.py

```diff
@@ -706,11 +706,48 @@ def overrideStorageClass(self, storageClass: str | StorageClass) -> DatasetRef:
             A new dataset reference that is the same as the current one but
             with a different storage class in the `DatasetType`.
         """
+        return self.update(storage_class=storageClass)
+
+    def update(
```

Comment:
Maybe `replace()` to match the name used in `dataclasses`? `update` doesn't immediately suggest that a new version is being returned. I called the equivalent `ResourcePath.replace` for that reason.